docs/src/pages/cli.mdx

```diff
@@ -0,0 +1,259 @@
+---
+title: Marmot CLI
+---
+
+# Marmot CLI
+
+## Usage
+
+You can run Marmot like this:
+
+```bash
+marmot --config /path/to/config.toml -cluster-addr localhost:4221 -cluster-peers 'nats://localhost:4222/'
+```
+
+You can clean up Marmot triggers and changelogs like this:
+
+```bash
+marmot --config /path/to/config.toml --cleanup
+```
+
+You can take snapshot and upload like this:
+
+```bash
+marmot --config /path/to/config.toml --save-snapshot
+```
+
+Minimal Config
+
+```toml
+seq_map_path="/tmp/marmot-1-sm.cbor"
+db_path="/tmp/marmot-1.db"
+node_id=1
+```
+
+## Flags
+
+- `--config`: Path to the configuration file.
+- `--cleanup`: If set, only cleanup marmot triggers and changelogs.
+- `--save-snapshot`: If set, only take snapshot and upload.
+- `--cluster-addr`: Cluster listening address.
+- `--cluster-peers`: Comma separated list of clusters.
+- `--leaf-servers`: Comma separated list of leaf servers.
+
+## Configuration
+
+The configuration file is in TOML format and contains the following sections:
+
+### Snapshot Configuration
+
+- `enabled`: Enable snapshotting.
+- `interval`: Interval for taking snapshots.
+- `store`: Type of storage for snapshots. Options are "nats", "s3", "webdav", "sftp".
+- `nats`: NATS object store configuration.
+- `s3`: S3 configuration.
+- `webdav`: WebDAV configuration.
+- `sftp`: SFTP configuration.
+
+### Replication Log Configuration
+
+- `shards`: Number of shards.
+- `max_entries`: Maximum number of entries.
+- `replicas`: Number of replicas.
+- `compress`: Enable compression.
+- `update_existing`: Update existing entries.
+
+### NATS Configuration
+
+- `urls`: NATS server URLs.
+- `subject_prefix`: Subject prefix for NATS messages.
+- `stream_prefix`: Stream prefix for NATS messages.
+- `server_config`: Server configuration file for NATS.
+- `seed_file`: Seed file for NATS.
+- `user_name`: User name for NATS.
+- `user_password`: User password for NATS.
+- `ca_file`: CA file for NATS.
+- `cert_file`: Certificate file for NATS.
+- `key_file`: Key file for NATS.
+- `bind_address`: Bind address for NATS.
+- `connect_retries`: Number of connection retries for NATS.
+- `reconnect_wait_seconds`: Wait time in seconds before reconnecting to NATS.
+
+### Logging Configuration
+
+- `verbose`: Enable verbose logging.
+- `format`: Logging format.
+
+### Prometheus Configuration
+
+- `bind`: Bind address for Prometheus.
+- `enable`: Enable Prometheus.
+- `namespace`: Namespace for Prometheus.
+- `subsystem`: Subsystem for Prometheus.
+
+
+
+## Sample Config File
```

Comment:
Not sure if copy pasting a sample config is good idea when there is sample config file that is part of package.

Reply:
I think it's much better for the doc, but agree copy & paste is not a good idea. Will try to find if there's way to include inline file in mdx.